app/controllers/users/reset_passwords_controller.rb

```diff
@@ -21,19 +21,19 @@ def create
     end
 
     def edit
-      result = PasswordResetTokenValidator.new(token_user).submit
-
-      analytics.password_reset_token(**result.to_h)
-      irs_attempts_api_tracker.forgot_password_email_confirmed(
-        success: result.success?,
-        failure_reason: irs_attempts_api_tracker.parse_failure_reason(result),
-      )
+      if FeatureManagement.redirect_to_clean_edit_password_url?
+        session[:reset_password_token] = params[:reset_password_token]
```

Comment:
I think we may want delete the key/value if there's a form submission where the token is successfully used or the token is otherwise invalid.